pkg/api/unversioned/group_version.go

```diff
@@ -48,9 +48,18 @@ type GroupVersion struct {
 	Version string
 }
 
+func (gv GroupVersion) IsEmpty() bool {
+	return len(gv.Group) == 0 && len(gv.Version) == 0
+}
+
 // String puts "group" and "version" into a single "group/version" string. For the legacy v1
 // it returns "v1".
 func (gv GroupVersion) String() string {
+	// special case the internal apiVersion for kube
```

Comment:
I think the "foo/" string serialization is temporary until these refactors are complete. The internal version should never be serialized to a string in my opinion. "foo/" let's us split, get two tokens, and avoid more special cases